File: pagertabsindicator/src/main/java/com/hold1/pagertabsindicator/TabViewProvider.java
package com.hold1.pagertabsindicator;

import android.net.Uri;
import android.view.View;

/**
 * Created by Cristian Holdunu on 08/11/2017.
 */

public class TabViewProvider {

    public interface ImageProvider {
        Uri getImageUri(int position);
    }
    
    public interface CustomView {
        View getView(int position);
    }

}

File: pagertabsindicator/src/main/java/com/hold1/pagertabsindicator/TabViewProvider.kt
package com.hold1.pagertabsindicator

import android.net.Uri
import android.view.View

sealed class TabViewProvider() {

    interface ImageProvider {
        fun getImageUri(position: Int): Uri?
        fun getImageResourceId(position: Int): Int
    }

    interface CustomView {
        fun getView(position: Int): View
    }
}